app/assets/javascripts/application.js

```diff
@@ -4,6 +4,32 @@
 //= require bootstrap-sprockets
 //= require activestorage
 //= require cable
+//= require gritter
+
+
+var IconOnclick = function () {
+  $('.clickable').click(function () {
+    var parent = $(this).closest('li')
+    parent.hide();
+    parent.find('#delete_item_' + $(this).data('id')).val(true);
+  });
+}
+
+var InputOnchange = function () {
+  $('.num_input').on('change', function () {
+    var max = parseInt($(this).attr('max'));
+    var min = parseInt($(this).attr('min'));
+    var quantity = parseInt($(this).val());
+    var price = parseInt($(this).data('price'));
+    var parent = $(this).closest('li');
+    parent.find('.text_right.price').text('$' + quantity * price);
```

Comment:
js thuần những phép tính toán nó có nhiều cái đưa ra kết quả theo 10*16 số thập phân a nha. Viết như này sẽ có bug đó a. ví dụ : `0.1 * 0.4 = 0.04000000000000001`